op-challenger/fault/resolver.go

```diff
@@ -0,0 +1,57 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// ContractCaller is a minimal interface of [ethclient.Client] to allow for mocking.
+type ContractCaller interface {
+	CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+}
+
+type Resolver struct {
+	ContractCaller
+	fdgAddr common.Address
+	fdgAbi  *abi.ABI
+}
+
+// NewResolver creates a new [Resolver] instance.
+func NewResolver(client ContractCaller, fdgAddr common.Address) (*Resolver, error) {
+	fdgAbi, err := bindings.FaultDisputeGameMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	return &Resolver{
+		client,
+		fdgAddr,
+		fdgAbi,
+	}, nil
+}
+
+// buildResolveData creates the transaction data for the Resolve function.
+func (r *Resolver) buildResolveData() ([]byte, error) {
+	return r.fdgAbi.Pack("resolve")
+}
+
+// CallResolve executes an eth_call to the Resolve function.
+func (r *Resolver) CallResolve(ctx context.Context) (bool, error) {
+	resolveData, err := r.buildResolveData()
+	if err != nil {
+		return false, err
+	}
+	_, err = r.CallContract(ctx, ethereum.CallMsg{
+		To:   &r.fdgAddr,
+		Data: resolveData,
+	}, nil)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
```

Comment:
Shouldn't this be going via the tx manager like when we call claims? It seems to be a very different pattern to how we're doing calls in `faultResponder`.

Given `faultResponder` already has the ability to build attack, defend and step calls it seems like maybe we should be just adding the ability to call `resolve` in there so we have one place that's responsible for dispute game contract calls?

Reply:
I'm not 100% sure if we should be going through the tx manager for performing the `eth_call`, but we should be using `buildResolveData()` and then doing an `eth_call`. Maybe we extend the txmanager to be able to do that `eth_call` b/c it already has the L1 connection or we can pass in a L1 client from somewhere.

Reply:
Refactored the `eth_call` to pass through the `TxManager` in #6371 - let me know what you think there. Will close this PR out now since it's sufficiently outdated.